python-checks/src/main/java/org/sonar/python/checks/RobustCipherAlgorithmCheck.java

```diff
@@ -63,14 +86,54 @@ public class RobustCipherAlgorithmCheck extends PythonSubscriptionCheck {
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, subscriptionContext -> {
       CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
-      Symbol calleeSymbol = callExpr.calleeSymbol();
-      if (calleeSymbol != null) {
-        String fqn = calleeSymbol.fullyQualifiedName();
-        if (fqn != null && sensitiveCalleeFqns.contains(fqn)) {
-          subscriptionContext.addIssue(callExpr.callee(), MESSAGE);
-        }
-      }
+      Optional.ofNullable(callExpr)
+        .map(CallExpression::calleeSymbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(fqn -> sensitiveCalleeFqns.contains(fqn) || SSL_SET_CIPHERS_FQN.equals(fqn))
+        .ifPresent(str -> addMethodSpecificIssue(subscriptionContext, callExpr, str));
     });
   }
 
+  private static void addMethodSpecificIssue(
```

Comment:
I feel the naming is not really representative of what happens here. It is not an issue for a specific method. It is just an issue. I think this could be refactored in the previous ifPresent. This would allow the removal of a cognitive overhead with two contradicting `if`s in a row. `!SSL_SET_CIPHERS_FQN.equals(fullyQualifiedName)` and `SSL_SET_CIPHERS_FQN.equals(fqn)`.

Reply:
I removed this method in 1dd32b3d33381b16d208e2d0d833a5b7c2956532.

Reply:
I removed this method in 1dd32b3d33381b16d208e2d0d833a5b7c2956532.